types/module/simulation/simulation.go

```diff
@@ -0,0 +1,97 @@
+package simulation
+
+import (
+	"math/rand"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/regen-network/regen-ledger/types/module/server"
+)
+
+// AppModuleSimulation defines the standard functions that every module should expose
+// for the SDK blockchain simulator
+type AppModuleSimulation interface {
+	// randomized genesis states
+	GenerateGenesisState(input *module.SimulationState)
+
+	// content functions used to simulate governance proposals
+	ProposalContents(simState module.SimulationState) []simulation.WeightedProposalContent
+
+	// randomized module parameters for param change proposals
+	RandomizedParams(r *rand.Rand) []simulation.ParamChange
+
+	// register a func to decode the each module's defined types from their corresponding store key
+	RegisterStoreDecoder(sdk.StoreDecoderRegistry)
+
+	// simulation operations (i.e msgs) with their respective weight
+	WeightedOperations(simState module.SimulationState) []simulation.WeightedOperation
+}
+
+// AppSimulationManager defines a simulation manager that provides the high level utility
+// for managing and executing simulation functionalities for a group of modules
+type AppSimulationManager struct {
+	Modules       []module.AppModuleSimulation // array of app modules; we use an array for deterministic simulation tests
+	StoreDecoders sdk.StoreDecoderRegistry     // functions to decode the key-value pairs from each module's store
+	manager       *server.Manager
+}
+
+// NewAppSimulationManager creates a new SimulationManager object
+//
+// CONTRACT: All the modules provided must be also registered on the module Manager
+func NewAppSimulationManager(serverManager *server.Manager, modules ...module.AppModuleSimulation) *AppSimulationManager {
+	return &AppSimulationManager{
+		Modules:       modules,
+		StoreDecoders: make(sdk.StoreDecoderRegistry),
+		manager:       serverManager,
```

Comment:
could we just pass in the result of `serverManager.GetWeightedOperationsHandlers()` instead of the whole `serverManager`?